src/core/include/openvino/op/constant.hpp

```diff
@@ -670,6 +682,50 @@ class OPENVINO_API Constant : public Op {
         }
     }
 
+    template <element::Type_t Type,
+              typename T,
+              typename StorageDataType = fundamental_type_for<Type>,
+              typename std::enable_if<Type == element::Type_t::nf4 && std::is_integral<T>::value, bool>::type = true>
+    void write_buffer(const std::vector<T>& source) {
+        auto p = get_data_ptr_nc<Type>();
+        size_t i = 0;
+        for (; i < source.size() / 2; i++) {
+            const auto v1 = value_in_range<Type>(source[i * 2]) & 0x0F;
+            const auto v2 = value_in_range<Type>(source[i * 2 + 1]) & 0x0F;
+            const auto v = (v2 << 4) | v1;
+            p[i] = static_cast<StorageDataType>(v);
+        }
+        if (source.size() % 2) {
+            const auto v = value_in_range<Type>(source[i * 2]) & 0x0F;
+            p[i] = static_cast<StorageDataType>(v);
+        }
+    }
+
+    template <element::Type_t Type,
+              typename T,
+              typename StorageDataType = fundamental_type_for<Type>,
+              typename std::enable_if<Type == element::Type_t::nf4 &&
+                                          (std::is_floating_point<T>::value || std::is_same<T, bfloat16>::value ||
+                                           std::is_same<T, float16>::value),
+                                      bool>::type = true>
+    void write_buffer(const std::vector<T>& source) {
+        auto p = get_data_ptr_nc<Type>();
+        size_t i = 0;
+        for (; i < source.size() / 2; i++) {
+            const auto idx1 = ConvertNF4::quantize(static_cast<float>(source[i * 2]));
```

Comment:
Just detail if there will be any change consider rename, the `idx` name as it is not index but quantized value.

Reply:
Ok, in the next PR I will change it. Is name like qnt1, qnt2 appropriate ?